### ISSUE TITLE
Invalid inventoryfile structure can cause hard to debug error

### ISSUE THREAD

**Issue by donoghuc** [puppetlabs/bolt#2430]
## Describe the Bug
With an inventory like:
```yaml
groups:
  name: ssh
  targets:
    - name: nix_target
      uri: example.com
  config:
    transport: ssh
    ssh:
      user: root
      password: password
```
This was hard to diagnose:
```console
NoMethodError: undefined method `key?' for ["name", "ssh"]:Array
  /home/cas/working_dir/bolt/lib/bolt/inventory/group.rb:27:in `initialize'
  /home/cas/working_dir/bolt/lib/bolt/inventory/group.rb:72:in `new'
  /home/cas/working_dir/bolt/lib/bolt/inventory/group.rb:72:in `block in initialize'
  /home/cas/working_dir/bolt/lib/bolt/inventory/group.rb:72:in `map'
  /home/cas/working_dir/bolt/lib/bolt/inventory/group.rb:72:in `initialize'
  /home/cas/working_dir/bolt/lib/bolt/inventory/inventory.rb:25:in `new'
  /home/cas/working_dir/bolt/lib/bolt/inventory/inventory.rb:25:in `initialize'
  /home/cas/working_dir/bolt/lib/bolt/inventory.rb:87:in `new'
  /home/cas/working_dir/bolt/lib/bolt/inventory.rb:87:in `create_version'
  /home/cas/working_dir/bolt/lib/bolt/inventory.rb:77:in `from_config'
  /home/cas/working_dir/bolt/lib/bolt/cli.rb:62:in `inventory'
  /home/cas/working_dir/bolt/lib/bolt/cli.rb:410:in `execute'
  /home/cas/working_dir/bolt/exe/bolt:11:in `<top (required)>'
  /home/cas/working_dir/bolt/vendor/bundle/ruby/2.5.0/bin/bolt:23:in `load'
  /home/cas/working_dir/bolt/vendor/bundle/ruby/2.5.0/bin/bolt:23:in `<top (required)>'
```
## Expected Behavior
Error message with helpful message instead of stack trace.

## Additional Context
Bandaid incomplete solution: https://github.com/puppetlabs/bolt/pull/2416

**Comment by beechtom** [comment]
We actually have a ticket on the board that will address this! :) https://github.com/puppetlabs/bolt/issues/2413

**Comment by beechtom** [comment]
Closed by #2440

**Comment by beechtom** [comment]
Closed by #2440
